### PR TITLE
Feature/codegen str concat

### PR DESCRIPTION
**Pull request by damniko**
Ladies and gentlemen: String concatenation has arrived!!!

ArduinoC already supports concatenation using the '+' operator on String objects, so the process is muy simple.
If the (plus) expression has the type string, there are 3 cases to consider:

1. Both operands are strings, in which case there is no problem - just emit both operands with a '+' between them!
2. The left operand is _not_ a string, in which case it can be converted to one: `String(left)`
3. The right operand is _not_ a string, in which case we convert that to a string as above

It has been tested and is working (for simple cases at least) on an actual Arduino - wub wub!

### CODE REVIEW

**Comment by damniko** [comment]
[Code generation: String concatenation](https://trello.com/c/7FqaVwRb/91-code-generation-string-concatenation)

File: STEP/CodeGeneration/CodeGenerationVisitor.cs [hunk 1]
```diff
@@ -29,7 +34,19 @@ private void EmitAppend(string line)
     private void EmitAppend(TypeVal typeVal)
     {
         // TODO: translate our types to Arduino types here
-        _stringBuilder.Append(typeVal.ToString()+" ");
+        switch(typeVal)
+        {
+            case TypeVal.String:
+                _stringBuilder.Append("String ");
```
Comment (WassawRoki) [inline STEP/CodeGeneration/CodeGenerationVisitor.cs]: string med stort men ikke bool og double, er det bare sådan arduino er ?

**Review by WassawRoki — APPROVED** [review]
nice

File: STEP/CodeGeneration/CodeGenerationVisitor.cs (hunk 1 above)
Comment (damniko) [reply STEP/CodeGeneration/CodeGenerationVisitor.cs]: Yes - der er en `string` datatype i Arduino/C++, men den er meget basic (stort set det samme som et char array som i gode gamle C). 
`String` (med stort S) er en klasse de selv har lavet til Arduino, hvor de har lavet operator overloading for string concatenation, equality, etc.
Der står lidt om det her: https://www.arduino.cc/reference/en/language/variables/data-types/string/

Comment (WassawRoki) [reply STEP/CodeGeneration/CodeGenerationVisitor.cs]: oright, det sprang bare i øjenene, men epic nok, måske en comment jeg ville aldrig have gættet det :P

**Review by selectionator — APPROVED** [review]
LGTM

**Pull request merged**